Tests/Unit/Twig/DescriptionExtensionTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2016 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\ResourceBundle\Tests\Unit\Twig;
+
+use Symfony\Cmf\Component\Resource\Description\DescriptionFactory;
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
+use Symfony\Cmf\Component\Resource\Description\Description;
+use Symfony\Cmf\Bundle\ResourceBundle\Twig\DescriptionExtension;
+
+class DescriptionExtensionTest extends \PHPUnit_Framework_TestCase
+{
+    private $descriptionFactory;
+    private $extension;
+    private $resource;
+    private $description;
+
+    public function setUp()
+    {
+        $this->descriptionFactory = $this->prophesize(DescriptionFactory::class);
+        $this->extension = new DescriptionExtension(
+            $this->descriptionFactory->reveal()
+        );
+        $this->resource = $this->prophesize(CmfResource::class);
+        $this->description = $this->prophesize(Description::class);
+    }
+
+    /**
+     * It should provide the cmf_resource_description function.
+     */
+    public function testProvideResourceDescription()
+    {
+        $this->descriptionFactory->getPayloadDescriptionFor($this->resource->reveal())->willReturn($this->description->reveal());
+        $function = $this->getFunction('cmf_resource_description');
+        $this->assertNotNull($function);
+        $callable = $function->getCallable();
+        $description = $callable($this->resource->reveal());
+
+        $this->assertInstanceOf(Description::class, $description);
+        $this->assertSame($this->description->reveal(), $description);
```

Comment:
I don't like this test very much. What about converting it to a functional test? (unit testing it doesn't make much sense, as the unit doesn't contain logic)

```php
// ...
class DescriptionExtensionTest extends \PHPUnit_Framework_TestCase
{
    private $twig;
    private $descriptionFactory;

    protected function setUp()
    {
        $this->descriptionFactory = $this->prophesize(DescriptionFactory::class);

        $this->twig = new \Twig_Environment(new \Twig_Loader_Array([]));
        $this->twig->addExtension(new DescriptionExtension($this->descriptionFactory->reveal()));
    }

    public function testProvidesResourceDescription()
    {
        $resource = $this->prophesize(CmfResource::class)->reveal();

        $description = $this->prophesize(Description::class);
        $description->get('title')->willReturn('Hello world!');

        $this->descriptionFactory->getPayloadDescriptionFor($resource)->willReturn($description->reveal());

        $template = $this->twig->createTemplate('{{ cmf_resource_description(resource).get("title") }}');
        $result = $template->render([
            'resource' => $resource,
        ]);

        $this->assertEquals('Hello world!', $result);
    }
}
```